staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer.go

```diff
@@ -61,11 +64,28 @@ func (s *LocationStreamer) InputStream(ctx context.Context, apiVersion, acceptHe
 	if transport == nil {
 		transport = http.DefaultTransport
 	}
+	requestAddress := s.Location.Host
+	klog.Infof("Sending request to %q.", requestAddress)
+
+	if httpTransport, ok := transport.(*http.Transport); ok {
+		networkContext := server.NetworkContext{ConnectivityServiceName: "cluster"}
+		contextDialer, err := server.Lookup(networkContext)
```

Comment:
can we plumb the dialer resolver to the streamer, rather than a singleton map?

Reply:
Not sure I am interpreting your question correctly. The streamer is just for log support. This pattern will need to be replicated to the various other outbound requests. (Eg. webhooks, aggregator, etcd, ...)

Reply:
Plumbed in the second commit.